OpenRA.Mods.Common/Widgets/Logic/Settings/DisplaySettingsLogic.cs

```diff
@@ -269,16 +269,14 @@ Func<bool> InitPanel(Widget panel)
 			};
 
 			var colorManager = modData.DefaultRules.Actors[SystemActors.World].TraitInfo<ColorPickerManagerInfo>();
-			colorManager.Color = ps.Color;
 
-			var colorDropdown = panel.Get<DropDownButtonWidget>("PLAYERCOLOR");
-			colorDropdown.IsDisabled = () => worldRenderer.World.Type != WorldType.Shellmap;
-			colorDropdown.OnMouseDown = _ => ColorPickerLogic.ShowColorDropDown(colorDropdown, colorManager, worldRenderer, () =>
-			{
-				Game.Settings.Player.Color = colorManager.Color;
-				Game.Settings.Save();
-			});
-			colorDropdown.Get<ColorBlockWidget>("COLORBLOCK").GetColor = () => ps.Color;
+			SettingsUtils.BindColorWidget(panel, worldRenderer, colorManager, () => worldRenderer.World.Type != WorldType.Shellmap, "PLAYERCOLOR", Game.Settings.Player, "Color");
+
+			bool IsColorPickerDisabled() => !Game.Settings.Game.UsePlayerStanceColors || worldRenderer.World.Type != WorldType.Shellmap;
+			SettingsUtils.BindColorWidget(panel, worldRenderer, colorManager, IsColorPickerDisabled, "SELFCOLOR", gs, "SelfColor");
```

Comment:
Use `nameof()` for the "preference" field names.

Why not pass them as `ref Color color` though, and save the reflection?

Reply:
You can't use ref in local functions / delagates

Reply:
nameof(gs.SelfColor) is "gs.SelfColor"